### DISCUSSION TITLE
EBSD map to Young's modulus

### DISCUSSION THREAD

**Discussion by bibiboom89** [mtex-toolbox/mtex#577] [Ask Anything] [answered]
Hi everyone,

I have a pretty simple issue, but I surprisingly can't find the solution anywhere.

I have an EBSD map stored in a CTF file that I load with:
```
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');
```

I would like for each pixel of the EBSD map, calculate the Young modulus along the three directions X, Y, Z for a given stiffness tensor  Cij (hexagonal crystal) :
```
C1111=162.4; C3333=180.7; C1122=92; C1133=69; C1313=46.7;
C2222=C1111; C2233=C1133; C2323=C1313; C1212=(C1111-C1122)/2;
Cij=[C1111 C1122 C1133 0 0 0
     C1122 C2222 C2233 0 0 0
     C1133 C2233 C3333 0 0 0
     0     0     0     C2323 0 0
     0     0     0     0 C1313 0
     0     0     0     0 0 C1212 ];
```

And then plot the young modulus map, with a colorbar aside. Something like:
`figure; plot(ebsd('Ti'),ebsd('Ti').YoungModulus); colorbar`

If anyone could help me with that, that would be much appreciated :)

Also, bonus question: which are X, Y, Z-axis? Z is normal to the EBSD plan? How to plot the axis on the figure?

Thank you in advance for your support,
Rémy

**Comment by kilir** [comment, marked as answer]
Hi,
here's a simple example:
```matlab
mtexdata forsterite

% file name
fname = fullfile(mtexDataPath,'tensor','Olivine1997PC.GPa');

% define the tensor in crystal coordinates
C = stiffnessTensor.load(fname,ebsd('f').CS)

% derive the crystal direction parallel to your reference direction
yC = inv(ebsd('f').orientations).*yvector;

% compute the youngsmodulus parallel to those crystal directions
yM = YoungsModulus(C,yC)

%plot it
plot(ebsd('f'),yM)
```



Have a look here: https://mtex-toolbox.github.io/AnisotropicTheory.html for some hints.

Bonus answer: x,y,z are specimen coordinates, hence the ones you use to define a vector3d or the ones you use to map your crystal directions to.
 z is always normal to your screen. Where x(N,E,S,W) and z(in or out) and accordingly y point to on your screen is defined by the plotting conventions you set in your preferences.

Hope that helps.

Cheers,
Rüdiger

**Reply by bibiboom89** [reply]
Thank you a lot. 
I did look at the tutorial you shared, but the part I was missing is: 
`yC = inv(ebsd('f').orientations).*yvector;`

Is there a way to plot the axis system on the map, next to the scale? And the colorbar on the side?

**Reply by bibiboom89** [reply]
I found how to add a color bar to the figure (but not the axis system), but I have an issue adding a legend on it.
Using your exemple as illustration, here is my code to plot my figure:
```
figure
plot(ebsd('f'),yM*10^3)
cl=colorbar('eastoutside'); cl.Label.String='Title Test';
```
In attachment is what I got with the forsterite data of your example. As you can see, the title is outside the box and therefore not visible.

Also, with my own data, the scale is cropped (2 is missing in front 00) and I don't know how to fix it (see file in attachment).

Thank you in advance for your help,
Rémy

![My_file](https://user-images.githubusercontent.com/63070505/99598630-d2fda100-29c7-11eb-855c-fae1ba78d7c1.png)
![Forsterite_test](https://user-images.githubusercontent.com/63070505/99598641-d7c25500-29c7-11eb-8e63-2cf748fd67f9.png)

**Comment by ralfHielscher** [comment]
Hi Remi,

as for the colorbar have a look here

https://mtex-toolbox.github.io/Annotations.html

What do you mean by "axis system"?
In your figure also have a look into the MTEX menu it has a submenu called
annotations maybe you find there what you want.



Ralf.


********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Reply by bibiboom89** [reply]
OK, there is a special mtex colorbar. 
I will look more deeply into that. 

Thank you a lot for your support and for developing MTEX which is an excellent tool.

Have a great day.

**Reply by Savillian** [reply]
Hi everyone,

Instead of just using the x, y, or z-vector, how could you evaluate the crystal orientations for any 3D vector in the specimen reference frame?

All the best,
Alec

**Comment by bjoerneske** [comment]
Alec you can define any vector 3d
E.g.

Vector= vector3d (1,2,5)

And use that

man. 3. mai 2021, 18:28 skrev AlphaTwister ***@***.***>:

**Reply by Savillian** [reply]
Thanks bjoerneske! I was looking to create a large series of random orientations for evaluating properties, but I finally found the command I was looking for in the MTEX documentation (vector3d.rand).
